metricbeat/mb/module/wrapper.go

```diff
@@ -243,27 +250,17 @@ func (msw *metricSetWrapper) String() string {
 }
 
 func (msw *metricSetWrapper) Test(d testing.Driver) {
-	done := make(chan struct{})
 	d.Run(msw.Name(), func(d testing.Driver) {
-		// ReportingMetricSet would hang out forever, perhaps we can add a timeout based test in the future
-		if _, ok := msw.MetricSet.(mb.ReportingMetricSet); ok {
-			d.Warn("test", "metricset doesn't support testing")
-			return
-		}
-
-		reporter := &testingReporter{
-			driver: d,
-			done:   done,
-		}
-		msw.fetch(reporter)
+		events := make(chan beat.Event, 1)
+		done := receiveOneEvent(d, events, msw.module.maxStartDelay+5*time.Second)
```

Comment:
Hardcoding a timeout is not a best practise. Please make it configurable in a config file perhaps.

Reply:
There is a configurable timeout associated with each module. We should ensure that value instead and ensure the value is initialized with a default. Do you want to open a PR to change this?

https://github.com/elastic/beats/blob/bc4b07c922d57bdafd3bc12e9e23f190c2592a09/metricbeat/mb/mb.go#L287